x/programs/rust/wasmlanche_sdk/src/host/mod.rs

```diff
@@ -50,21 +51,24 @@ pub unsafe fn store_bytes(
 /// Gets the length of the bytes associated with the key from the host.
 ///
 /// # Safety
-/// The caller must ensure that key_ptr + key_len points to valid memory locations.
-pub unsafe fn get_bytes_len(ctx: &Context, key_ptr: *const u8, key_len: usize) -> i32 {
+/// The caller must ensure that `key_ptr` + `key_len` points to valid memory locations.
+#[must_use]
+pub unsafe fn get_bytes_len(ctx: Context, key_ptr: *const u8, key_len: usize) -> i32 {
```

Comment:
`Context` is `Copy` which means that on 64-bit architecture, copying directly from the stack is going to be faster than pointer operations. I'm not quite sure about 32-bit architecture since, technically, a pointer is smaller than the data, but the pointer indirection could cause a slowdown that's greater than copying the extra 32-bits. 

Happy to change this back if you (the reviewer) thinks I should.

Reply:
i think passing `Context` is better